packages/core/core/src/services/session-manager.ts

```diff
@@ -141,6 +148,12 @@ class SessionManager {
     deviceId: string,
     origin: string
   ): Promise<{ token: string; sessionId: string }> {
+    if (typeof this.provider?.deleteExpiredByIdentifier === 'function') {
```

Comment:
Do we actually need this? We're doing the expired cleanup handling in the next line. It seems like we can remove all of the deleteExpired changes in this PR and make it simpler, unless I'm missing something.